src/toil/job.py

```diff
@@ -3026,3 +3030,32 @@ def convertPromises(kwargs):
                 foundPromisedRequirement = True
         return foundPromisedRequirement
 
+
+class JobStoreFile(str):
```

Comment:
To qualify for the name JobStoreFile, I should be able to do more things with it. This still is an ID—a glorified one, sure—but the name will conflict if we actually want create an full-blown class that wholly represents a file in the job store.

I also dislike that the "jobStore" part leaks to the user. The job store should be hidden. In the context of the file store I would have preferred to refer to file IDs as just fileID. This naming choice solidifies that mistake.

How about FileID or FileRef or FileInfo?